conans/model/build_info.py

```diff
@@ -55,22 +55,24 @@ def _sorted_components(self):
         Sorted components from less dependent to the most one
         :return: ordered list of components
         """
-        # Sort first elements with less items in .deps attribute
-        comps = sorted(self._components.values(), key=lambda component: len(component.deps))
         # Save name of unsorted elements
-        unsorted_names = [comp.name for comp in comps]
-
+        unsorted_names = list(self._components.keys())
         sorted_comps = []
+        element = unsorted_names[0]
         while unsorted_names:
```

Comment:
Probably it is even easier to not use a `unsorted_names` and iterate while the return has not all the elements (not tested):

```
ordered = []
while len(ordered) != len(self._components):
    # Search for next element to be processed
    for comp in self._components:
        if comp in ordered:
            continue
        # check if all the deps are declared
        if not all([dep in self._components.keys() for dep in comp.deps])):
            raise ConanException("Component %s not found in cpp_info object" % e)
        # check if all the deps are already added to ordered
        if all([dep in ordered.keys() for dep in comp.deps])): 
            break
    else:
         raise ConanException("Probably this means a loop? add a test")

    ordered.add(comp)

return ordered
```

Reply:
Seems to not work completely and fixing includes more complexity than the proposed solution. Not sure what you mean by "loop check". Do you think this could get stuck in an infinite loop?

Reply:
I meant that no matter the implementation you need also to be able to detect them. So you need a test. I think maybe my code above was able to detect it in the exception I'm raising.